shared/js/sidebar.jsx

```diff
@@ -0,0 +1,179 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var loop = loop || {};
+
+loop.sidebar = (function(mozL10n) {
+  "use strict";
+
+  var ROOM_STATES = loop.store.ROOM_STATES;
+  var sharedActions = loop.shared.actions;
+  var sharedUtils = loop.shared.utils;
+  var sharedMixins = loop.shared.mixins;
+
+  var SidebarControllerView = React.createClass({
+    mixins: [sharedMixins.UrlHashChangeMixin,
+             sharedMixins.DocumentLocationMixin,
```

Comment:
The above two mixins are unused. `Backbone.Events` should also be unused if my comment below is taken into consideration.